drivers/usb/device/usb_dc_mcux_nxp.h

```diff
@@ -0,0 +1,372 @@
+/*
+ * Copyright (c) 2018 NXP Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "usb_spec.h"
+
+/******************************************************************************
+ * Definitions
+ *****************************************************************************/
+/* KHCI instance count */
+#ifdef CONFIG_USB_DC_NXP_KHCI
+    #define USB_DEVICE_CONFIG_KHCI (1U)
+#else
+    #define USB_DEVICE_CONFIG_KHCI (0U)
+#endif
+
+/* EHCI instance count */
+#ifdef CONFIG_USB_DC_NXP_EHCI
+    #define USB_DEVICE_CONFIG_EHCI (1U)
+#else
+    #define USB_DEVICE_CONFIG_EHCI (0U)
+#endif
+
+/* IP3511 FS instance count */
+#ifdef CONFIG_USB_DC_NXP_IP3511FS
+    #define USB_DEVICE_CONFIG_LPCIP3511FS (1U)
+#else
+    #define USB_DEVICE_CONFIG_LPCIP3511FS (0U)
+#endif
+
+/* IP3511 HS instance count */
```

Comment:
Why do we need any of these instance count macros?

Reply:
@MaureenHelm For example: USB_DEVICE_CONFIG_EHCI, I convert CONFIG_USB_DC_NXP_EHCI to USB_DEVICE_CONFIG_EHCI in the head file.
In MCUXpresso SDK USB controller driver, USB_DEVICE_CONFIG_EHCI is used to define the EHCI controller driver instance count (if it is 0, the function is disable), In Zephyr USB device stack, the multiple instances is not supported yet, CONFIG_USB_DC_NXP_EHCI only means enable or disable. In order to support multiple instance in furture and do less changes of original MCUX USB device controller driver (usb_device_ehci.c), It may be better to keep this convertion.

Reply:
Ok, this makes more sense after you moved `usb_dc_mcux_nxp.h` to `ext/`